kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/src/main/java/org/eclipse/kura/demo/bnd/heater/Heater.java

```diff
@@ -0,0 +1,305 @@
+
+    private final ScheduledExecutorService worker;
+    private ScheduledFuture<?> handle;
+
+    private float temperature;
+    private Map<String, Object> properties;
+    private final Random random;
+
+    // ----------------------------------------------------------------
+    //
+    // Dependencies
+    //
+    // ----------------------------------------------------------------
+
+    public Heater() {
+        super();
+        this.random = new Random();
+        this.worker = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    @Reference(
+            name = "CloudService",
+            service = CloudService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.STATIC,
+            unbind = "unsetCloudService"
+    )
+    public void setCloudService(CloudService cloudService) {
+        this.cloudService = cloudService;
+    }
+
+    public void unsetCloudService(CloudService cloudService) {
+        this.cloudService = null;
+    }
+
+    // ----------------------------------------------------------------
+    //
+    // Activation APIs
+    //
+    // ----------------------------------------------------------------
+
+    @Activate
+    protected void activate(Map<String, Object> properties) {
+        logger.info("Activating Heater...");
+
+        this.properties = properties;
+        for (String s : properties.keySet()) {
+            logger.info("Activate - " + s + ": " + properties.get(s));
+        }
+
+        // get the mqtt client for this application
+        try {
+
+            // Acquire a Cloud Application Client for this Application
+            logger.info("Getting CloudClient for {}...", APP_ID);
+            this.cloudClient = this.cloudService.newCloudClient(APP_ID);
+            this.cloudClient.addCloudClientListener(this);
+
+            // Don't subscribe because these are handled by the default
+            // subscriptions and we don't want to get messages twice
+            doUpdate(false);
+        } catch (Exception e) {
+            logger.error("Error during component activation", e);
+            throw new ComponentException(e);
+        }
+        logger.info("Activating Heater... Done.");
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        logger.debug("Deactivating Heater...");
+
+        // shutting down the worker and cleaning up the properties
+        this.worker.shutdown();
+
+        // Releasing the CloudApplicationClient
+        logger.info("Releasing CloudApplicationClient for {}...", APP_ID);
+        this.cloudClient.release();
+
+        logger.debug("Deactivating Heater... Done.");
+    }
+
+    @Modified
+    public void updated(Map<String, Object> properties) {
+        logger.info("Updated Heater...");
+
+        // store the properties received
+        this.properties = properties;
+        for (String s : properties.keySet()) {
+            logger.info("Update - " + s + ": " + properties.get(s));
```

Comment:
I just spotted this one... Should better be `Update - {}: {}` ... like a few lines above.